### PR TITLE
introduce zio-nio and zio-nio-core submodules

### PR DESCRIPTION
**Pull request by jczuchnowski**
Part of #121 .

1. splits project into two submodules zio-nio and zio-nio-core (directories nio/ and nio-core/)
2. restores unopinionated implementations of:

- AsynchronousChannnel
- AsynchronousChannnelGroup
- AsynchronousFileChannel
- FileChannel
- SelectableChannel

### CODE REVIEW

**Comment by pshemass** [comment]
fix the build and it's good to go

**Comment by jczuchnowski** [comment]
Yes, I need to fix the documentation...

**Comment by mtsokol** [comment]
Great to see this! I just noticed this PR, can't wait to see a release for that!

File: nio-core/src/main/scala/zio/nio/core/channels/Channel.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package zio.nio.core.channels
+
+import java.nio.channels.{ Channel => JChannel }
+
+import zio.{ IO, UIO }
+
+trait Channel {
+  protected val channel: JChannel
+
+  final private[channels] val close: IO[Exception, Unit] =
```
Comment (domsj) [inline nio-core/src/main/scala/zio/nio/core/channels/Channel.scala]: this one should be publicly accessible?

Comment (jczuchnowski) [reply nio-core/src/main/scala/zio/nio/core/channels/Channel.scala]: How did I miss this... :astonished:

**Comment by jczuchnowski** [comment]
@pshemass @domsj done. I'll merge it if you see no other issues. Then we can work on what's left.

**Pull request merged**